### ISSUE TITLE
Can't pass logging handlers to EventLog.handlers_list

### ISSUE THREAD

**Issue by yuvipanda** [jupyter/telemetry#14]
Even with #12, I can't pass a list of logging handlers to handlers_list. I get:

```
[E 2019-07-09 16:43:30.034 JupyterHub app:2492]
    Traceback (most recent call last):
      File "/home/yuvipanda/code/jupyterhub/jupyterhub/app.py", line 2489, in launch_instance_async
        await self.initialize(argv)
      File "/home/yuvipanda/code/jupyterhub/jupyterhub/app.py", line 2131, in initialize
        self.eventlog = EventLog(parent=self)
      File "/home/yuvipanda/code/telemetry/jupyter_telemetry/eventlog.py", line 55, in __init__
        super().__init__(*args, **kwargs)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/config/configurable.py", line 84, in __init__
        self.config = config
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/traitlets.py", line 585, in __set__
        self.set(obj, value)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/traitlets.py", line 574, in set
        obj._notify_trait(self.name, old_value, new_value)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/traitlets.py", line 1139, in _notify_trait
        type='change',
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/traitlets.py", line 1176, in notify_change
        c(change)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/traitlets.py", line 819, in compatible_observer
        return func(self, change)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/config/configurable.py", line 186, in _config_changed
        self._load_config(change.new, traits=traits, section_names=section_names)
      File "/home/yuvipanda/.local/share/virtualenvs/telemetry/lib/python3.7/site-packages/traitlets/config/configurable.py", line 153, in _load_config
        setattr(self, name, deepcopy(config_value))
      File "/usr/lib/python3.7/copy.py", line 150, in deepcopy
        y = copier(x, memo)
      File "/usr/lib/python3.7/copy.py", line 215, in _deepcopy_list
        append(deepcopy(a, memo))
      File "/usr/lib/python3.7/copy.py", line 180, in deepcopy
        y = _reconstruct(x, memo, *rv)
      File "/usr/lib/python3.7/copy.py", line 280, in _reconstruct
        state = deepcopy(state, memo)
      File "/usr/lib/python3.7/copy.py", line 150, in deepcopy
        y = copier(x, memo)
      File "/usr/lib/python3.7/copy.py", line 240, in _deepcopy_dict
        y[deepcopy(key, memo)] = deepcopy(value, memo)
      File "/usr/lib/python3.7/copy.py", line 169, in deepcopy
        rv = reductor(4)
    TypeError: can't pickle _thread.RLock objects
    
ERROR:asyncio:Task exception was never retrieved
```

With the following config:

```python
import logging
c.JupyterHub.authenticator_class = 'jupyterhub.auth.DummyAuthenticator'
c.JupyterHub.spawner_class = 'jupyterhub.spawner.SimpleLocalProcessSpawner'

c.EventLog.handlers_list = lambda: [logging.StreamHandler()]
c.EventLog.allowed_schemas = [
    'hub.jupyter.org/server-action'
]
```

I suspect this is not caught in the tests somehow because we aren't reading the config in from files? unsure...

**Comment by Zsailer** [comment]
I'm learning a lot about traitlets here… 

When config is loaded from a file, it’s *first* stored in a Config object. Then, it's transferred to the real object by making a *deepcopy* of the value.

This transfering step is the problem. The original Config object doesn’t actually enforce the traittype before deep copying. As a result, the function-wrapping I added in #12 has no effect. I'm exploring a new way around the issue. I'll keep you updated.

**Comment by Zsailer** [comment]
I think this is solved in #17
